File: NC/string-encode-and-decode.py
class Solution:
    def encode(self, strs: List[str]) -> str:
        newList = []
        for s in strs:
            newList.append(f"{len(s)}#")
            newList.append(s)

        ans = "".join(newList)

        print(ans)

        return ans

    def decode(self, s: str) -> List[str]:

        res = []

        i = 0
        while i < len(s):
            numString = ""

            while s[i] != "#":
                numString += s[i]
                i += 1

            strLength = int(numString)

            i += 1
            # if strLength!=0:
            res.append(s[i:i+strLength])
            i += strLength

        return res

File: NC/string-encode-and-decode.cpp
class Solution
{
public:
    string encode(vector<string> &strs)
    {
        string ans;

        for (auto s : strs)
        {
            ans += (to_string(s.size()) + "#" + s);
        }

        return ans;
    }

    vector<string> decode(string s)
    {
        vector<string> ans;

        for (int i = 0; i < s.size();)
        {
            int j = i;
            string length;
            while (s[j] != '#')
            {
                length += s[j];
                j++;
            }
            int lengthInt = stoi(length);
            string word = s.substr(j + 1, lengthInt);
            ans.push_back(word);
            i = j + 1 + stoi(length);
        }

        return ans;
    }
};
